include/revng/ADT/TrackingContainer.h

```diff
@@ -76,6 +82,11 @@ class TrackingContainer {
                              ToReturn.InspectedKeys.begin()));
 
     ToReturn.Exact = Exact.isSet();
+    if (Exact.isSet())
+      for (auto &Entry : Content) {
+        using KOT = KeyedObjectTraits<value_type>;
+        ToReturn.InspectedKeys.insert(KOT::key(Entry));
+      }
```

Comment:
i do it because i find out that is easier for some parts of the code on being able to see that we rely on all paths by enumerating all of them, instead of pushing the if that checks if the container is Exact in all those locations.

Reply:
That doesn't sound safe, you'd be ignoring the fact that you're also relying on the *non*-existence of al those *not* listed.

Reply:
i thought about this, i think this is correct and they are two different things.
From a high level point of view we will have a visit that returns all read existing read fields of a model, and then a visit tha give a set of paths will tell you which ones added a entry to a vector that was marked as exact. This is the only implementation that makes sense now that all tracking primitives are private.

Reply:
what do you mean? it is not irrelevant. If you are considering a diff that is just a removal you don't care about exactness, you only care about the list all read nodes.

Reply:
I still see this in first commit. Please review the history before asking for reviews.

Reply:
Ping

Reply:
if i understood correctly your concern this should be gone. i did `git reset --soft` are recreated manually each commit